File: 0953-reverse-only-letters/0953-reverse-only-letters.java
class Solution {
    public String reverseOnlyLetters(String s) {
        char[] arr = s.toCharArray();  // Convert string to char array
        int left = 0, right = arr.length - 1; // Initialize pointers

        while (left < right) {
            if (!Character.isLetter(arr[left])) {
                left++;  // Move left pointer if not a letter
            } else if (!Character.isLetter(arr[right])) {
                right--; // Move right pointer if not a letter
            } else {
                // Swap letters
                char temp = arr[left];
                arr[left++] = arr[right];
                arr[right--] = temp;
            }
        }

        return new String(arr);  // Convert char array back to string
    }
}

File: 0953-reverse-only-letters/0953-reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        arr=list(s)
        left,right=0,len(arr)-1
        while left<right:
            if not arr[left].isalpha():
                left+=1
            elif not arr[right].isalpha():
                right-=1
            else:
                arr[left],arr[right]=arr[right],arr[left]
                left+=1
                right-=1
        return ''.join(arr)
